packages/backend-defaults/src/entrypoints/auditor/Auditor.ts

```diff
@@ -0,0 +1,310 @@
+    request?: AuditorEventRequest;
+  } & AuditorEventStatus,
+];
+
+/** @public */
+export const defaultProdFormat = winston.format.combine(
+  winston.format.timestamp({
+    format: 'YYYY-MM-DD HH:mm:ss',
+  }),
+  winston.format.errors({ stack: true }),
+  winston.format.splat(),
+  winston.format.json(),
+);
+
+/**
+ * Adds `isAuditorEvent` field
+ *
+ * @public
+ */
+export const auditorFieldFormat = winston.format(info => {
+  return { ...info, isAuditorEvent: true };
+})();
+
+/** @public */
+export interface RootAuditorOptions {
+  meta?: JsonObject;
+  format?: Format;
+  transports?: winston.transport[];
+}
+
+/**
+ * A {@link @backstage/backend-plugin-api#AuditorService} implementation based on winston.
+ *
+ * @public
+ */
+export class DefaultAuditorService implements AuditorService {
+  private readonly impl: DefaultRootAuditorService;
+  private readonly auth: AuthService;
+  private readonly httpAuth: HttpAuthService;
+  private readonly plugin: PluginMetadataService;
+
+  private constructor(
+    impl: DefaultRootAuditorService,
+    deps: {
+      auth: AuthService;
+      httpAuth: HttpAuthService;
+      plugin: PluginMetadataService;
+    },
+  ) {
+    this.impl = impl;
+    this.auth = deps.auth;
+    this.httpAuth = deps.httpAuth;
+    this.plugin = deps.plugin;
+  }
+
+  /**
+   * Creates a {@link DefaultAuditorService} instance.
+   */
+  static create(
+    impl: DefaultRootAuditorService,
+    deps: {
+      auth: AuthService;
+      httpAuth: HttpAuthService;
+      plugin: PluginMetadataService;
+    },
+  ): DefaultAuditorService {
+    return new DefaultAuditorService(impl, deps);
+  }
+
+  private async log<TMeta extends JsonObject>(
+    options: AuditorEventOptions<TMeta>,
+  ): Promise<void> {
+    const auditEvent = await this.reshapeAuditorEvent(options);
+    this.impl.log(auditEvent);
+  }
+
+  async createEvent<TMeta extends JsonObject>(
+    options: Parameters<AuditorCreateEvent<TMeta>>[0],
+  ): ReturnType<AuditorCreateEvent<TMeta>> {
+    if (!options.suppressInitialEvent) {
+      await this.log({ ...options, status: 'initiated' });
+    }
+
+    return {
+      success: async params => {
+        // return undefined if both objects are empty; otherwise, merge the objects
+        const meta =
+          Object.keys(options.meta ?? {}).length === 0 &&
+          Object.keys(params?.meta ?? {}).length === 0
+            ? undefined
+            : { ...options.meta, ...params?.meta };
+
+        await this.log({
+          ...options,
+          meta,
+          status: 'succeeded',
+        });
+      },
+      fail: async params => {
+        // return undefined if both objects are empty; otherwise, merge the objects
+        const meta =
+          Object.keys(options.meta ?? {}).length === 0 &&
+          Object.keys(params.meta ?? {}).length === 0
+            ? undefined
+            : { ...options.meta, ...params.meta };
+
+        await this.log({
+          ...options,
+          ...params,
```

Comment:
why failed function need params but success function not pass params to log function